src/doc/tarpl/exotic-sizes.md

```diff
@@ -0,0 +1,118 @@
+% Exotically Sized Types
+
+Most of the time, we think in terms of types with a fixed, positive size. This
+is not always the case, however.
+
+
+
+
+
+# Dynamically Sized Types (DSTs)
+
+Rust also supports types without a statically known size. On the surface, this
+is a bit nonsensical: Rust *must* know the size of something in order to work
+with it! DSTs are generally produced as views, or through type-erasure of types
+that *do* have a known size. Due to their lack of a statically known size, these
+types can only exist *behind* some kind of pointer. They consequently produce a
+*fat* pointer consisting of the pointer and the information that *completes*
+them.
+
+For instance, the slice type, `[T]`, is some statically unknown number of
+elements stored contiguously. `&[T]` consequently consists of a `(&T, usize)`
+pair that specifies where the slice starts, and how many elements it contains.
+Similarly, Trait Objects support interface-oriented type erasure through a
+`(data_ptr, vtable_ptr)` pair.
+
+Structs can actually store a single DST directly as their last field, but this
+makes them a DST as well:
+
+```rust
+// Can't be stored on the stack directly
+struct Foo {
+    info: u32,
+    data: [u8],
+}
+```
+
+**NOTE: As of Rust 1.0 struct DSTs are broken if the last field has
+a variable position based on its alignment.**
+
+
+
+
+
+# Zero Sized Types (ZSTs)
+
+Rust actually allows types to be specified that occupy *no* space:
+
+```rust
+struct Foo; // No fields = no size
+
+// All fields have no size = no size
+struct Baz {
+    foo: Foo,
+    qux: (),      // empty tuple has no size
+    baz: [u8; 0], // empty array has no size
+}
+```
+
+On their own, ZSTs are, for obvious reasons, pretty useless. However as with
```

Comment:
I kinda like having ZST in the title for CTRL+F, but I understand the sentiment.